src/Orleans.Transactions/InClusterTM/TransactionManager.cs

```diff
@@ -5,10 +5,99 @@
 using System.Threading;
 using Microsoft.Extensions.Options;
 using Microsoft.Extensions.Logging;
+using Orleans.Runtime;
 using Orleans.Transactions.Abstractions;
+using Orleans.Runtime.Configuration;
 
 namespace Orleans.Transactions
 {
+    internal class TransactionManagerMetrics
+    {
+        private const string StartTransactionRequestTPS = "TransactionManager.StartTransaction.TPS";
+        private const string AbortTransactionRequestTPS = "TransactionManager.AbortTransaction.TPS";
+        private const string CommitTransactionRequestTPS = "TransactionManager.CommitTransaction.TPS";
+       
+        private const string AbortedTransactionDueToDependencyTPS = "Transaction.AbortedDueToDependency.TPS";
+
+        private const string AbortedTransactionDueToMissingInfoInTransactionTableTPS =
+            "Transaction.AbortedDueToMissingInfoInTransactionTable.TPS";
+        private const string AbortedTransactionTPS = "Transaction.Aborted.TPS";
+        //pending transaction produced TPS in the monitor window
+        private const string PendingTransactionTPS = "Transaction.Pending.TPS";
+        //TPS for validated transaction in the monitor window
+        private const string ValidatedTransactionTPS = "Transaction.Validated.TPS";
+
+        internal long StartTransactionRequestCounter { get; set; }
```

Comment:
are you accessing `long` values directly instead of exposing (safer?) increment method for performance reason?

Reply:
This TransactionManager class is used only in TransactionManagerGrain, so it is single threaded. hence I wasn't concerned by threading issues.

Reply:
These either need to be fields which are incremented via `Interlocked.Increment` or we need to guard them via a lock. The `TransactionManager` class might be misleading: it sits in a folder called `InClusterTM`, but it's actually used by both grain & external TM. `++` wont give us accurate results - it might in some cases spike the result by skipping the reset to 0.

Reply:
I'm not concerned about threading issues here, it just feel not natural to deal directly with long here. I think having IncrementXXX method would be a better design. This way someone cannot decrement the values...